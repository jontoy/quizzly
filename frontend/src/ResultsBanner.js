import React from "react";

const ResultsBanner = ({ numberCorrect, totalQuestions }) => {
  const score = numberCorrect / totalQuestions;
  let color;
  let topMessage;
  let bottomMessage;
  let textColor = "light";
  if (score >= 0.9) {
    color = "success";
    topMessage = "Congratulations!";
    bottomMessage = "You really know your stuff!";
  } else {
    if (score >= 0.5) {
      color = "info";
      topMessage = "So close!";
      bottomMessage = "You almost had it!";
    } else {
      color = "warning";
      topMessage = "Too bad...";
      bottomMessage = "Review the material and give it another shot!";
      textColor = "dark";
    }
  }
  return (
    <div className={`ResultsBanner jumbotron bg-${color} text-${textColor}`}>
      <h1 className="display-4">{topMessage}</h1>
      <p className="lead">
        You got {numberCorrect} out of {totalQuestions} correct on this attempt.
      </p>
      <hr className="my-4" />
      <p>{bottomMessage}</p>
    </div>
  );
};
export default ResultsBanner;
