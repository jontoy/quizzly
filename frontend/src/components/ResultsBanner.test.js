import React from "react";
import { render } from "@testing-library/react";
import ResultsBanner from "./ResultsBanner";

it("renders without crashing", () => {
  render(<ResultsBanner />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<ResultsBanner />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when user scores moderately", () => {
  const { asFragment } = render(
    <ResultsBanner numberCorrect={3} totalQuestions={5} />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when user scores well", () => {
  const { asFragment } = render(
    <ResultsBanner numberCorrect={19} totalQuestions={20} />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when user scores badly", () => {
  const { asFragment } = render(
    <ResultsBanner numberCorrect={1} totalQuestions={5} />
  );
  expect(asFragment()).toMatchSnapshot();
});
