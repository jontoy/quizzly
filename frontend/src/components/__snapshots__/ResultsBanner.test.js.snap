// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ResultsBanner jumbotron bg-warning text-dark"
  >
    <h1
      class="display-4"
    >
      Too bad...
    </h1>
    <p
      class="lead"
    >
      You got 0 out of 0 correct on this attempt.
    </p>
    <hr
      class="my-4"
    />
    <p>
      Review the material and give it another shot!
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when user scores badly 1`] = `
<DocumentFragment>
  <div
    class="ResultsBanner jumbotron bg-warning text-dark"
  >
    <h1
      class="display-4"
    >
      Too bad...
    </h1>
    <p
      class="lead"
    >
      You got 1 out of 5 correct on this attempt.
    </p>
    <hr
      class="my-4"
    />
    <p>
      Review the material and give it another shot!
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when user scores moderately 1`] = `
<DocumentFragment>
  <div
    class="ResultsBanner jumbotron bg-info text-light"
  >
    <h1
      class="display-4"
    >
      So close!
    </h1>
    <p
      class="lead"
    >
      You got 3 out of 5 correct on this attempt.
    </p>
    <hr
      class="my-4"
    />
    <p>
      You almost had it!
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when user scores well 1`] = `
<DocumentFragment>
  <div
    class="ResultsBanner jumbotron bg-success text-light"
  >
    <h1
      class="display-4"
    >
      Congratulations!
    </h1>
    <p
      class="lead"
    >
      You got 19 out of 20 correct on this attempt.
    </p>
    <hr
      class="my-4"
    />
    <p>
      You really know your stuff!
    </p>
  </div>
</DocumentFragment>
`;
